webapi/Controllers/PluginController.cs

```diff
@@ -0,0 +1,110 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Threading.Tasks;
+using CopilotChat.WebApi.Auth;
+using CopilotChat.WebApi.Hubs;
+using CopilotChat.WebApi.Models.Storage;
+using CopilotChat.WebApi.Options;
+using CopilotChat.WebApi.Storage;
+using CopilotChat.WebApi.Utilities;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.SignalR;
+using Microsoft.Extensions.Logging;
+using Microsoft.SemanticKernel.Diagnostics;
+
+namespace CopilotChat.WebApi.Controllers;
+
+/// <summary>
+/// Controller responsible for returning the service options to the client.
+/// </summary>
+[ApiController]
+public class PluginController : ControllerBase
+{
+    private const string PluginStateChanged = "PluginStateChanged";
+    private readonly ILogger<PluginController> _logger;
+    private readonly IDictionary<string, Plugin> _availablePlugins;
+    private readonly ChatSessionRepository _sessionRepository;
+
+    public PluginController(
+        ILogger<PluginController> logger,
+        IDictionary<string, Plugin> availablePlugins,
+        ChatSessionRepository sessionRepository)
+    {
+        this._logger = logger;
+        this._availablePlugins = availablePlugins;
+        this._sessionRepository = sessionRepository;
+    }
+
+    /// <summary>
+    /// Fetches a plugin's manifest.
+    /// </summary>
+    /// <param name="manifestDomain">The domain of the manifest.</param>
+    /// <returns>The plugin's manifest JSON.</returns>
+    [HttpGet]
+    [Route("getPluginManifest")]
```

Comment:
We don't need "Get" in the route since it's a GET request. Also, pluginManifest should be plural since it's a collection whose Id is specified by manifestDomain. So the route should be `pluginManifests/{domain}` where domain is what is currently in the query.

Reply:
@alliscode this method isn't loading the plugin manifest from a stored collection. It's just making a fetch call to the URL domain that's passed it, essentially, loading the JSON from a webpage, so maybe something like `domains/{manifestDomain}/pluginManifest` would make more sense